crates/egui/src/containers/collapsing_header.rs

```diff
@@ -62,11 +61,23 @@ impl CollapsingState {
         self.state.open = open;
     }
 
-    pub fn toggle(&mut self, ui: &Ui) {
+    pub fn toggle_open(&mut self, ui: &Ui) {
         self.state.open = !self.state.open;
         ui.ctx().request_repaint();
     }
 
+    pub fn is_hidden(&self) -> bool {
+        self.state.hidden
+    }
+
+    pub fn toggle_hidden(&mut self) {
+        self.state.hidden = !self.state.hidden;
+    }
+
+    pub fn set_hidden(&mut self, hidden: bool) {
+        self.state.hidden = hidden;
+    }
```

Comment:
What does it mean for a `CollapsingHeader` to be hidden?

Reply:
Well windows can become hidden, collapsing headers won't do anything but I will update the PR, so this code won't be accessible when not using a window. I also have made the same event option for normal widgets to expand/collapse which I will include in this PR as well.

Reply:
I think I will let hidden in the struct for now, otherwise it requires:

A:  Use of generics which I think is an absolute overkill in this instance and ugly.
B: Big restructure of code which I'm willing to do but I will have to take my time for that PR.

Reply:
I build a split between the two states on this branch: https://github.com/Norlock/egui/commit/40b1fb773bd3c327affb49614a505116d45be74b#diff-817170c50650af7962ada22bb6459b20eeb54ce921043e4db8522368ccf8933e. I don't think its a real improvement but you can check it out